### PR TITLE
Allow setting docker runtime from YBuild for CppGTests

### PR DESCRIPTION
**Pull request by shirl-resonai**
Allow setting docker runtime from YBuild for CppGTests

### CODE REVIEW

File: yabt/buildcontext.py [hunk 1]
```diff
@@ -379,6 +380,12 @@ def run_in_buildenv(
         container_work_dir = PurePath('/project')
         if work_dir:
             container_work_dir /= work_dir
+        if runtime:
+            docker_run.extend([
+                '--runtime', runtime
+            ])
+            logger.info('Running with runtime {}', runtime)
```
Comment (bergden-resonai) [inline yabt/buildcontext.py]: Do we need the print?

Comment (itamaro) [reply yabt/buildcontext.py]: better remove this log. we log the entire command before running it.
```suggestion
```

File: yabt/buildcontext.py [hunk 2]
```diff
@@ -379,6 +380,12 @@ def run_in_buildenv(
         container_work_dir = PurePath('/project')
         if work_dir:
             container_work_dir /= work_dir
+        if runtime:
+            docker_run.extend([
+                '--runtime', runtime
```
Comment (itamaro) [inline yabt/buildcontext.py]:
```suggestion
                '--runtime', runtime,
```

File: yabt/builders/cpp.py [hunk 3]
```diff
@@ -368,7 +371,9 @@ def cpp_gtest_tester(build_context, target):
     test_cmd.extend(target.props.test_flags)
     build_context.run_in_buildenv(
         # TODO: target.props.in_testenv,
-        target.props.in_buildenv, test_cmd, target.props.test_env)
+        target.props.in_buildenv, test_cmd, target.props.test_env,
+        None, True, target.props.runtime
```
Comment (itamaro) [inline yabt/builders/cpp.py]: better use keyword arg, and let the func signature take care of default values for other args

```suggestion
        runtime=target.props.runtime
```

**Comment by shirl-resonai** [comment]
> what's the use case that requires overriding Docker runtime for `CppGTests`? if there's a need for custom runtimes, shouldn't it be more generic, such that it can be controlled for all builder types?

I am running with nvidia-docker. This might apply to one or two other types of builder, but for most builders it isn't relevant - it has to be one where ybt does the `docker run`.

File: yabt/builders/cpp.py [hunk 4]
```diff
@@ -190,6 +192,7 @@ def cpp_app_builder(build_context, target):
     CPP_SIG + [
         ('test_flags', PT.StrList, None),  # flags to append to test command
         ('test_env', None),  # env vars to inject in test process
+        ('runtime', PT.str, None)
```
Comment (itamaro) [inline yabt/builders/cpp.py]:
```suggestion
        ('runtime', PT.str, None),
```

File: yabt/builders/cpp.py [hunk 5]
```diff
@@ -145,7 +145,9 @@ def get(self, param, config, target, fallback):
 
 
 register_app_builder_sig(
-    'CppApp', [('executable', PT.File, None), ('main', PT.Target, None)])
```
Comment (itamaro) [inline yabt/builders/cpp.py]: what changed here?

Comment (shirl-resonai) [reply yabt/builders/cpp.py]: Nothing, I can roll this back

**Pull request merged**